assets/src/scripts/components/graph/view/index.js

```diff
@@ -1,86 +1,161 @@
+ * @param  {Object} store     Redux store
+ * @param  {String} chartType Kind of chart
+ */
+const drawClipPath = function (elem, store, chartType) {
+    elem.append('defs')
+        .append('clipPath')
+            .attr('id', `${chartType}-clip-path`)
+            .append('rect')
+                .call(link(store, (rect, chartPosition) => {
+                    rect.attr('x', 0)
+                        .attr('y', 0)
+                        .attr('width', chartPosition.width + FOCUS_CIRCLE_RADIUS)
+                        .attr('height', chartPosition.height);
+                }, getChartPosition(chartType)));
+};
+
 /**
  * Draws a water-levels graph.
  * @param  {Object} store   Redux store
  * @param  {Object} node    DOM node to draw graph into
  * @param  {Object} options {agencycode, siteid} of site to draw
  */
-export default function (elem, store) {
-    elem.call(link(store, (elem, waterLevels) => {
-            elem.classed('loading', !waterLevels || !waterLevels.samples)
-                .classed('has-error', waterLevels && waterLevels.error);
-        }, getCurrentWaterLevels))
-        .append('div')
-            .classed('graph-container', true)
-            .call(link(store, drawAxisYLabel, createStructuredSelector({
-                unit: getCurrentWaterLevelUnit
-            })))
-            .call(link(store, drawTooltip, createStructuredSelector({
-                cursorPoint: getCursorPoint,
-                unit: getCurrentWaterLevelUnit
+export const drawChart = function (elem, store, chartType) {
+    elem.append('g')
+        .classed('chart', true)
+        .classed(chartType, true)
+        .call(drawClipPath, store, chartType)
+        .call(link(store, (elem, pos) => {
+            elem.attr('transform', `translate(${pos.x}, ${pos.y})`);
+        }, getChartPosition(chartType)))
+        .call(link(store, drawWaterLevels, createStructuredSelector({
+            lineSegments: getLineSegments,
+            xScale: getScaleX(chartType),
+            yScale: getScaleY(chartType)
+        })))
+        .call(callIf(chartType === 'main', link(store, drawAxisY, createStructuredSelector({
+            yScale: getScaleY(chartType),
+            cropSvgNode: () => chartType === 'main' ? elem : null,
+            containerSize: getContainerSize
+        }))))
+        .call(callIf(chartType === 'main', link(store, drawAxisX, createStructuredSelector({
+            xScale: getScaleX(chartType),
+            layout: getChartPosition(chartType)
+        }))))
+        .call(link(store, drawFocusLine, createStructuredSelector({
+            cursor: getCursor,
+            xScale: getScaleX(chartType),
+            yScale: getScaleY(chartType)
+        })))
+        .call(link(store, drawFocusCircle, createStructuredSelector({
+            cursorPoint: getCursorDatum,
+            xScale: getScaleX(chartType),
+            yScale: getScaleY(chartType)
+        })))
+        .call(g => {
```

Comment:
I'd still like to see the tooltip code factored into it's own draw function. There are a three separate pieces here that implement the tooltip function and I believe it would be clearer if they were segregated into their own function.

Reply:
👍 Yeah, these two could be moved pretty easily into a separate one. The tooltip text itself would be harder, since it's not part of the SVG.

Reply:
One thing I like about keeping it as-is, is that the state-dependent parts are all in `view/index.js`, and the other modules just export rendering functions that don't know anything about the store.

Reply:
I do like that aspect (state dependent parts are only here). I think there needs to be a comment or some explanation of why we are creating each piece of the svg then. At least for the tooltip. I'm trying to think about someone coming into this for the first time, trying to figure out what all the pieces do.

Reply:
Good point.. I'll comment all the external rendering calls in that module.